### ISSUE TITLE
could you support attribute macro like rust?

### ISSUE THREAD

**Issue by Silentdoer** [dart-lang/language#1158]
for generate fromJson and toJson

**Comment by lrhn** [comment]
Probably not.
We are aware that Dart does not have a good meta-programming story (see, e.g, https://github.com/dart-lang/language/issues/631), but we currently do not have a good, workable solution to that.
Something as low-level as the Rust macros is probably not how I would do it ... but until we figure out an actual way, it's not impossible either.

**Comment by mateusfccp** [comment]
I agree that meta-programming in Dart is currently very painful. Looking forward to see what the Dart team will come up with!
